### PR TITLE
Fix mana doublers not working during payment

### PR DESCRIPTION
**Pull request by tool4ever**
Caused by the freeze changes in #4794

Maybe there's a better long term fix if you can find one?

### CODE REVIEW

**Comment by tehdiplomat** [comment]
Whats the repro?

**Comment by tool4ever** [comment]
Uhh basically any SA that you want to also pay partly with mana from the doubled parts:
image

To be precise, it looks like after/(if) you finish casting you will get what you're owed in the pool by then

**Comment by tehdiplomat** [comment]
I'm guessing the "true" resolution should be where we check if the stack is frozen and if we should ignore frozen. We should also resolve it for mana triggers.

**Comment by tehdiplomat** [comment]
Yea i'm confused why those wouldn't go straight through here:

https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/zone/MagicStack.java#L262-L285

**Comment by tool4ever** [comment]
While the stack is frozen trigger events are only collected until we reach a stable state again (that's probably simplifying things a bit but the main idea)

**Comment by tehdiplomat** [comment]
I'll try to look tonight for a better long term fix

File: forge-game/src/main/java/forge/game/trigger/TriggerHandler.java [hunk 1]
```diff
@@ -266,7 +266,7 @@ public final void runTrigger(final TriggerType mode, final Map<AbilityKey, Objec
 
         if (mode == TriggerType.Always) {
             runStateTrigger(runParams);
-        } else if (game.getStack().isFrozen() || holdTrigger) {
+        } else if ((game.getStack().isFrozen() || holdTrigger) && mode != TriggerType.TapsForMana && mode != TriggerType.ManaAdded) {
```
Comment (tehdiplomat) [inline forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:269]: Oh i see, we're not even getting to MagicStack before they get held.  We need to test if the trigger is a mana ability here to run it right away. We can merge for now and i'll try to clean it up soon.

**Pull request merged**